lib/compress/zstd_ldm.c

```diff
@@ -223,6 +223,17 @@ static U64 ZSTD_ldm_fillLdmHashTable(ldmState_t* state,
     return rollingHash;
 }
 
+void ZSTD_ldm_fillHashTable(
+            ldmState_t* state, const BYTE* ip,
+            const BYTE* iend, ldmParams_t const* params)
+{
+    U64 startingHash = ZSTD_rollingHash_compute(ip, params->minMatchLength);
+    ZSTD_ldm_fillLdmHashTable(
+        state, startingHash, ip, iend, state->window.base,
```

Comment:
Did the fuzzer find this bug? It still needs to be fixed.

Reply:
I'm having some trouble exposing this bug in fuzzer.c. I tried to get zstd to raise a seg fault by doing this but that didn't seem to work: 
1. Create a dict buffer of size 2*n
2. Create a src buffer of size n
3. Generate data of size n into src
4. memcpy n bytes from src to dict and dict + n
5. realloc(dict, n) 
Do you see a way to do it?

Reply:
The bug should be triggered by:
1. Pass in a dictionary of N bytes that has been malloc'd in a buffer with exactly N bytes.
2. Enable long range mode with the cdict (double check that this code is called).
3. Enable ASAN.

It would take quite unlucky (or engineered) circumstances to get the binary to crash without ASAN enabled, but ASAN should find it immediately.

Reply:
Ah perfect. I wasn't enabling ASAN. That did it